app/src/main/java/com/github/se/stepquest/map/Map.kt

```diff
@@ -1,14 +1,109 @@
+  var checkpointTitle by remember { mutableStateOf("") }
+
+  Scaffold(
+      content = {
+        Box(modifier = Modifier.fillMaxSize()) {
+          GoogleMap(
+              modifier = Modifier.fillMaxSize().testTag("GoogleMap"),
+          )
+          FloatingActionButton(
+              onClick = { showDialog = true },
+              modifier =
+                  Modifier.padding(16.dp)
+                      .align(Alignment.BottomEnd)
+                      .offset(y = (-96).dp)
+                      .size(48.dp)) {
+                Box(
+                    modifier =
+                        Modifier.size(48.dp) // Adjust the size of the icon container
+                            .background(Color.Blue, CircleShape), // Blue background
+                    contentAlignment = Alignment.Center) {
+                      Icon(
+                          painter = painterResource(R.drawable.map_marker),
+                          contentDescription = "Add checkpoint",
+                          tint = Color.Red)
+                    }
+              }
+        }
+      },
+      floatingActionButton = {
+        if (showDialog) {
+          AlertDialog(
+              onDismissRequest = { showDialog = false },
+              title = {
+                Row(verticalAlignment = Alignment.CenterVertically) {
+                  Text("New Checkpoint", modifier = Modifier.weight(1f)) // Title "New Checkpoint"
+                  IconButton(onClick = { showDialog = false }, modifier = Modifier.size(36.dp)) {
+                    Icon(
+                        Icons.Default.Close,
+                        contentDescription = "Close",
+                        tint = Color.Black // Black cancel button
+                        )
+                  }
+                }
+              },
+              text = {
+                Column(modifier = Modifier.padding(16.dp)) {
+                  Text(
+                      "Checkpoint name",
+                      modifier = Modifier.padding(bottom = 8.dp)) // Prompt "Checkpoint name"
+                  TextField(
+                      value = checkpointTitle,
+                      onValueChange = { checkpointTitle = it },
+                      label = { Text("Title") },
+                      modifier = Modifier.fillMaxWidth())
+                }
+              },
+              confirmButton = {
+                Box(
+                    modifier =
+                        Modifier.fillMaxWidth()
+                            .padding(horizontal = 16.dp)
+                            .height(48.dp) // Set the height of the button
+                    ) {
+                      Button(
+                          onClick = {
+                            // Here you can handle adding the checkpoint to the map
+                            val title = checkpointTitle
+                            showDialog = false
+                          },
+                          modifier =
```

Comment:
the color of the confirm button could be changed to blue to fit the color style of the app